### PR TITLE
Update Gradle wrapper to 7.1

### PR DESCRIPTION
**Pull request by breskeby**
- Fix new introduced deprecated usages
- Update to newer ospackage snapshot to include provided PR for fixing deprecated usage

This gradle release comes with improvements on incremental compilation which we should benefit from

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void apply(Project project) {
         File heapdumpDir = new File(project.getBuildDir(), "heapdump");
 
         project.getTasks().withType(Test.class).configureEach(test -> {
-            File testOutputDir = new File(test.getReports().getJunitXml().getDestination(), "output");
+            File testOutputDir = new File(test.getReports().getJunitXml().getOutputLocation().getAsFile().get(), "output");
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java:51]: getDestination has been deprecated

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/CheckstylePrecommitPlugin.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ public void execute(Task task) {
 
         project.getTasks().withType(Checkstyle.class).configureEach(t -> {
             t.dependsOn(copyCheckstyleConf);
-            t.reports(r -> r.getHtml().setEnabled(false));
+            t.reports(r -> r.getHtml().getRequired().set(false));
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/CheckstylePrecommitPlugin.java:97]: setEnabled has been deprecated

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/LoggerUsageTask.java [hunk 3]
```diff
@@ -62,8 +63,7 @@ public void setClasspath(FileCollection classpath) {
     @PathSensitive(PathSensitivity.RELATIVE)
     @SkipWhenEmpty
     public FileCollection getClassDirectories() {
-        return getProject().getConvention()
-            .getPlugin(JavaPluginConvention.class)
+        return getProject().getExtensions().getByType(JavaPluginExtension.class)
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/LoggerUsageTask.java:66]: access to convention api has been deprecated

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ThirdPartyAuditTask.java [hunk 4]
```diff
@@ -334,7 +334,7 @@ private String runForbiddenAPIsCli() throws IOException {
                 getProject().getConfigurations().getByName(CompileOnlyResolvePlugin.RESOLVEABLE_COMPILE_ONLY_CONFIGURATION_NAME)
             );
             spec.jvmArgs("-Xmx1g");
-            spec.setMain("de.thetaphi.forbiddenapis.cli.CliMain");
+            spec.getMainClass().set("de.thetaphi.forbiddenapis.cli.CliMain");
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ThirdPartyAuditTask.java:337]: getMain has been deprecated

File: build-tools/src/main/java/org/elasticsearch/gradle/LoggedExec.java [hunk 5]
```diff
@@ -55,7 +55,8 @@ public LoggedExec(FileSystemOperations fileSystemOperations) {
             doLast(new Action<Task>() {
                 @Override
                 public void execute(Task task) {
-                    if (LoggedExec.this.getExecResult().getExitValue() != 0) {
```
Comment (breskeby) [inline build-tools/src/main/java/org/elasticsearch/gradle/LoggedExec.java:58]: getExitValue() has been deprecated

File: distribution/packages/build.gradle [hunk 6]
```diff
@@ -48,7 +48,7 @@ buildscript {
     maven { url 'https://jitpack.io' }
   }
   dependencies {
-    classpath "com.github.breskeby:gradle-ospackage-plugin:ddb72a9922b934033827d48d296f7f3d470ac422"
+    classpath "com.github.breskeby:gradle-ospackage-plugin:98455c1"
```
Comment (breskeby) [inline distribution/packages/build.gradle:51]: This is a snaspshot reference to the ospackage plugin containing the changes of this provided PR https://github.com/nebula-plugins/gradle-ospackage-plugin/pull/400

File: distribution/packages/build.gradle [hunk 7]
```diff
@@ -124,10 +124,7 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
     String jdkString = jdk ? "" : "no-jdk-"
     String prefix = "${architecture == 'aarch64' ? 'aarch64-' : ''}${oss ? 'oss-' : ''}${jdk ? '' : 'no-jdk-'}${type}"
     destinationDirectory = file("${prefix}/build/distributions")
-
-    // SystemPackagingTask overrides default archive task convention mappings, but doesn't provide a setter so we have to override the convention mapping itself
-    conventionMapping.archiveFile = { objects.fileProperty().fileValue(file("${destinationDirectory.get()}/${packageName}-${project.version}-${jdkString}${archString}.${type}")) }
-
+    archiveFileName.value(project.provider({ "${packageName}-${project.version}-${jdkString}${archString}.${type}" } ))
```
Comment (breskeby) [inline distribution/packages/build.gradle:127]: conventionMapping access has been deprecated

**Review by pugnascotia — APPROVED** [review]
LGTM

**Pull request merged**